pageserver/src/tenant/mgr.rs

```diff
@@ -551,7 +550,22 @@ pub async fn detach_tenant(
     tenant_id: TenantId,
     detach_ignored: bool,
 ) -> Result<(), TenantStateError> {
-    detach_tenant0(conf, &TENANTS, tenant_id, detach_ignored).await
+    detach_tenant0(conf, &TENANTS, tenant_id, detach_ignored).await?;
+    let tmp_path = path_with_suffix_extension(conf.tenant_path(&tenant_id), TEMP_FILE_SUFFIX);
```

Comment:
Can we make detach_tenant0 pass back the temporary path?  That reduces the risk of making a mistake later if we ever change how temporary paths are calculated.

Reply:
Done